app/main/posts/modify/post-data-editor.controller.js

```diff
@@ -84,20 +86,51 @@ function PostDataEditorController(
         $scope.savePost();
     });
 
-    $scope.$on('event:edit:leave:form', function () {
-        if ($scope.parentForm.form && $scope.parentForm.form.$dirty) {
-            $scope.leavePost();
-        } else {
-            $scope.cancel();
+    $transitions.onStart({}, function (transition) {
```

Comment:
I don't really understand what this is doing?

Reply:
$transitions.onStart is a handler for when a transition starts. When a transition starts here you want to either let it go or cancel it here because you need the leavePost functionality with the warning modal and all that. So you have ignoreCancelEvent for a special case where you literally are transitioning after the user accepts (and so you don't want to show it again) and you have the transition.from().name making sure you were in a list.data.edit state (editing mode) , and if you are there, you either resolve or reject a promise depending on form state, user actions, etc. Resolve will let the transition continue. Reject will stop the transition meaning you stay in the edit mode
If you are not editing, you just return true to let the transition continue.